### PR TITLE
channel role added, allowed channels logic transparency

### PR DESCRIPTION
**Pull request by eugepae**

### CODE REVIEW

File: src/Softeq.NetKit.Chat.Data.Persistent.Sql/Migrations/ReworkChatMembersStructure.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+            Execute(@"ALTER TABLE [dbo].[ChannelMembers] ADD Role INT NULL");
+            Execute(@"UPDATE [dbo].[ChannelMembers] SET Role = 1 WHERE Role IS NULL");
+            Execute(@"ALTER TABLE [dbo].[ChannelMembers] ALTER COLUMN Role INT NOT NULL");
+        }
+
+        protected override void Down()
+        {
+            Execute(@"DROP PROCEDURE dbo.GetChannelsLastMessage");
+            Execute(@"ALTER TABLE [dbo].[ChannelMembers] DROP Role");
```
Comment (andrei-anfilets) [inline src/Softeq.NetKit.Chat.Data.Persistent.Sql/Migrations/ReworkChatMembersStructure.cs]: drop statement: ALTER TABLE [dbo].[ChannelMembers] DROP COLUMN Role

**Review by andrei-anfilets — DISMISSED** [review]

**Review by andrei-anfilets — DISMISSED** [review]

File: src/Softeq.NetKit.Chat.Data.Persistent.Sql/Repositories/ChannelMemberRepository.cs [hunk 2]
```diff
@@ -102,6 +105,72 @@ INNER JOIN
...
+                        m.{nameof(Message.Created)}, 
+                        m.{nameof(Message.Body)}, 
+                        m.{nameof(Message.ImageUrl)}, 
+                        m.{nameof(Message.Type)}, 
+                        m.{nameof(Message.OwnerId)}, 
+                        m.{nameof(Message.Updated)}, 
+                        m.{nameof(Message.AccessibilityStatus)},
+                        (SELECT COUNT(*)
+                         FROM Messages m
+                         WHERE m.{nameof(Message.ChannelId)} = @{nameof(channelId)} AND m.{nameof(Message.AccessibilityStatus)} = @accessibilityStatus AND m.{nameof(Message.Created)} < (
```
Comment (dzmitrymorozov) [inline src/Softeq.NetKit.Chat.Data.Persistent.Sql/Repositories/ChannelMemberRepository.cs]: Let's use nameof for accessibilityStatus param

File: src/Softeq.NetKit.Chat.Data.Persistent.Sql/Repositories/ChannelMemberRepository.cs [hunk 3]
```diff
@@ -102,6 +105,72 @@ INNER JOIN
...
+                             SELECT m.{nameof(Message.Created)}
+	                         FROM ChannelMembers cm
+	                         LEFT JOIN Messages m
+	                         ON m.{nameof(Message.Id)} = cm.{nameof(ChannelMember.LastReadMessageId)}
+	                         WHERE cm.{nameof(ChannelMember.MemberId)} = @{nameof(memberId)} AND cm.{nameof(ChannelMember.ChannelId)} = @{nameof(channelId)} AND m.{nameof(Message.AccessibilityStatus)} = @accessibilityStatus)) as {nameof(ChannelMemberAggregate.UnreadMessagesCount)}
```
Comment (dzmitrymorozov) [inline src/Softeq.NetKit.Chat.Data.Persistent.Sql/Repositories/ChannelMemberRepository.cs]: The same

File: src/Softeq.NetKit.Chat.Data.Persistent.Sql/Repositories/MessageRepository.cs [hunk 4]
```diff
@@ -234,35 +179,34 @@ public async Task<IReadOnlyCollection<Message>> GetMessagesWithOwnersAsync(Guid
...
-                        Messages message
-                    INNER JOIN Members member 
-                        ON message.{nameof(Message.OwnerId)} = member.{nameof(Member.Id)}
+                        Messages ms
+                    INNER JOIN Members m 
+                        ON ms.{nameof(Message.OwnerId)} = m.{nameof(Member.Id)}
                     WHERE
                         {nameof(Message.Created)} >= @{nameof(lastReadMessageCreated)}
                         AND {nameof(Message.ChannelId)} = @{nameof(channelId)} 
-                        AND message.{nameof(Message.AccessibilityStatus)} = @accessibilityStatus
```
Comment (dzmitrymorozov) [inline src/Softeq.NetKit.Chat.Data.Persistent.Sql/Repositories/MessageRepository.cs:265]: Let's add nameof for accessibilityStatus param

**Review by DmitryMorozov228 — APPROVED** [review]

**Pull request merged**